src/EFCore/Metadata/IForeignKey.cs

```diff
@@ -88,5 +88,42 @@ public interface IForeignKey : IReadOnlyForeignKey, IAnnotatable
         /// <typeparam name="TKey"> The type of key instances. </typeparam>
         /// <returns> A new factory. </returns>
         IDependentKeyValueFactory<TKey>? GetDependentKeyValueFactory<TKey>();
+
+        /// <summary>
+        ///     Finds the list of properties of this foreign key that are not also properties of another foreign key, if possible,
+        ///     otherwise returns all properties.
```

Comment:
```suggestion
        ///     Finds the list of properties of this foreign key that are not also properties of another foreign key.
```
The non-composite case should be filtered too, since the composite one can return an empty list. Let the caller deal with this, makes this method simpler.

Reply:
The most common case here is a single property. In this case we bail out early because the set of not overlapping properties is empty, and hence we want to just use all properties--in this case one.

The next most common case here is a composite property with no overlap. In this case we also want to use all properties.

Both these cases avoid allocating since there is no change to the properties that we will use.

The uncommon cases here are where there are overlapping properties. This is the only case for which we allocate new lists. However, as you point out, this list could be empty, in which case we again will default to all properties. I'll add code to check for this case.